unit_converter/__init__.py

```diff
@@ -0,0 +1,460 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import annotations
+
+import json
+import re
+import traceback
+from datetime import datetime
+from pathlib import Path
+from typing import Any
+from urllib.request import urlopen
+
+import albert
+import inflect
+import pint
+
+default_trigger = "convert "
+synopsis = "<amount> <from_unit> to <to_unit>"
+
+__doc__ = f"""
+Extension for converting units of length, mass, speed, temperature, time,
+current, luminosity, printing measurements, molecular substance, currency, and more
+
+Synopsis: {default_trigger}{synopsis}
+
+Examples:
+`{default_trigger}180 minutes to hrs`
+`{default_trigger}100 km to miles`
+`{default_trigger}88 mph to kph`
+`{default_trigger}32 degrees F to C`
+`{default_trigger}3.14159 rad to degrees`
+`{default_trigger}100 USD to EUR`
+"""
+
+md_iid = "0.5"
+md_version = "1.2"
+md_name = "Unit Converter"
+md_description = "Convert length, mass, temperature, time, currency, and more"
+md_license = "MIT"
+md_url = "https://github.com/albertlauncher/python"
+md_lib_dependencies = ["pint", "inflect"]
+md_maintainers = "@DenverCoder1"
+
+unit_convert_regex = re.compile(
+    r"(?P<from_amount>-?\d+\.?\d*)\s?(?P<from_unit>.*)\s(?:to|in)\s(?P<to_unit>.*)",
+    re.I,
+)
+
+units = pint.UnitRegistry()
+inflect_engine = inflect.engine()
+
+
+def load_config(config_path: Path) -> dict[str, Any]:
+    """
+    Strip comments and load the config from the config file.
+    """
+    with config_path.open("r") as config_file:
+        contents = config_file.read()
+    contents = re.sub(r"^\s*//.*$", "", contents, flags=re.MULTILINE)
+    return json.loads(contents)
+
+
+config_path = Path(__file__).parent / "config.jsonc"
```

Comment:
what exactly is in the config besides the rounding precision? Id rather drop rounding and output the most precise number we can get.